src/editor/Neovim/NeovimProtocol.re

```diff
@@ -27,53 +32,61 @@ let make = (nvimApi: NeovimApi.t) => {
     let _ =
       nvimApi.requestSync(
         "nvim_ui_attach",
-        Msgpck.List([
-          Msgpck.Int(20),
-          Msgpck.Int(20),
-          Msgpck.Map([
-            (Msgpck.String("rgb"), Msgpck.Bool(true)),
-            (Msgpck.String("ext_popupmenu"), Msgpck.Bool(true)),
-            (Msgpck.String("ext_tabline"), Msgpck.Bool(true)),
-            (Msgpck.String("ext_cmdline"), Msgpck.Bool(true)),
-            (Msgpck.String("ext_wildmenu"), Msgpck.Bool(true)),
-            (Msgpck.String("ext_linegrid"), Msgpck.Bool(true)),
-            /* (Msgpck.String("ext_multigrid"), Msgpck.Bool(true)), */
-            /* (Msgpck.String("ext_hlstate"), Msgpck.Bool(true)), */
+        M.List([
+          M.Int(20),
+          M.Int(20),
+          M.Map([
+            (M.String("rgb"), M.Bool(true)),
+            (M.String("ext_popupmenu"), M.Bool(true)),
+            (M.String("ext_tabline"), M.Bool(true)),
+            (M.String("ext_cmdline"), M.Bool(true)),
+            (M.String("ext_wildmenu"), M.Bool(true)),
+            (M.String("ext_linegrid"), M.Bool(true)),
+            /* (M.String("ext_multigrid"), M.Bool(true)), */
+            /* (M.String("ext_hlstate"), M.Bool(true)), */
           ]),
         ]),
       );
     ();
   };
 
-  let input = (key: string) => {
-    let _ =
-      nvimApi.requestSync("nvim_input", Msgpck.List([Msgpck.String(key)]));
-    ();
-  };
+  let input = (key: string) =>
+    nvimApi.requestSync("nvim_input", M.List([M.String(key)])) |> ignore;
 
   let bufAttach = id => {
-    let _ =
+    let _error =
       nvimApi.requestSync(
         "nvim_buf_attach",
-        Msgpck.List([Msgpck.Int(id), Msgpck.Bool(true), Msgpck.Map([])]),
+        M.List([M.Int(id), M.Bool(true), M.Map([])]),
       );
     ();
   };
 
-  let _ =
-    Event.subscribe(
-      nvimApi.onNotification,
-      n => {
-        let parsedEvent = Notification.parse(n.notificationType, n.payload);
+  Event.subscribe(
+    nvimApi.onNotification,
+    n => {
+      let parsedEvent = Notification.parse(n.notificationType, n.payload);
 
-        let f = n => {
-          Event.dispatch(onNotification, n);
-        };
+      let f = n => Event.dispatch(onNotification, n);
 
-        List.iter(f, parsedEvent);
-      },
-    );
+      List.iter(f, parsedEvent);
+    },
+  )
+  |> ignore;
+
+  let openFile = (~path=?, ~bufferId=?, ~openMethod=Buffer, ()) => {
+    let args =
+      switch (path, bufferId, openMethod) {
+      | (Some(p), None, Buffer) => M.List([M.String("edit" ++ " " ++ p)])
+      | (None, Some(id), Buffer) =>
+        M.List([M.String("edit" ++ " " ++ "#" ++ string_of_int(id))])
```

Comment:
Is there a reason this can't be combined to `"edit #"`?

Reply:
I was trying to be explicit about needing the `#` I didn't realise you needed that to open a buffer by id using `edit` thought some one might overlook it but having it like that makes it clear :thinking: that it was intentional although a comment would do the same and take any guess work out of that

Reply:
A comment sounds good! I think it would serve the same purpose. The concatenation isn't 'free' from a performance perspective, so unless there is a compelling reason to have it, it'd be better just to bake it in.